core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -550,12 +552,20 @@ trait EntityComponent {
 
       def actionForTypeName(workspaceContext: Workspace,
                             entityType: String,
-                            entityName: String
-      ): ReadAction[Seq[EntityAndAttributesResult]] =
+                            entityName: String,
+                            desiredFields: Set[AttributeName]
+      ): ReadAction[Seq[EntityAndAttributesResult]] = {
+        val formattedAttributePairs =
+          desiredFields.map(attributeName => s"('${attributeName.namespace}', '${attributeName.name}')").mkString(", ")
+        val attributesFilter =
+          if (desiredFields.isEmpty) "" else s"and (a.namespace, a.name) in ($formattedAttributePairs)"
+
         sql"""#${baseEntityAndAttributeSql(
             workspaceContext
-          )} where e.name = ${entityName} and e.entity_type = ${entityType} and e.workspace_id = ${workspaceContext.workspaceIdAsUUID}"""
+          )} where e.name = ${entityName} and e.entity_type = ${entityType} and e.workspace_id = ${workspaceContext.workspaceIdAsUUID}
+          #${attributesFilter}"""
```

Comment:
This is vulnerable to SQL injection. Try specifying a field of `')) OR 1=1; --` via the swagger UI (it'll be an error). The vulnerability here is because we're building a string literal and then using that literal via `#${attributesFilter}` instead of using bind parameters.

Take a look at WorkspaceComponent: https://github.com/broadinstitute/rawls/blob/5a23db18d962b2955a9ffca2240b054394aa6bcf/core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala#L214-L218 which does a very similar thing but uses bind params. I think you could implement it like this:

```scala
        // user requested specific attributes. include them in the where clause.
        val attrNamespaceNameTuples = reduceSqlActionsWithDelim(desiredFields.toSeq.map { attrName =>
          sql"(${attrName.namespace}, ${attrName.name})"
        })
        val attributesFilter =
          if (desiredFields.isEmpty) sql""
          else concatSqlActions(sql" and (a.namespace, a.name) in (", attrNamespaceNameTuples, sql")")

        concatSqlActions(
          sql"""#${baseEntityAndAttributeSql(
              workspaceContext
            )} where e.name = ${entityName} and e.entity_type = ${entityType} and e.workspace_id = ${workspaceContext.workspaceIdAsUUID}""",
          attributesFilter
        )
```

Reply:
I'll test it out!